source/PlayState.hx

```diff
@@ -8,32 +12,47 @@ import obstacle.Obstacle;
 import obstacle.ObstacleGenerator;
 import player.Player;
 
-class PlayState extends FlxState
-{
+class PlayState extends FlxState {
 	var player:Player;
 
+	var heroPower:HeroPower;
+
 	var obstacleGenerator:ObstacleGenerator<Obstacle>;
 	var deadlyObstacleGenerator:ObstacleGenerator<DeadlyObstacle>;
 
 	var SECONDS_PER_OBSTACLE(default, never):Float = .5;
 	var SECONDS_PER_DEADLY_OBSTACLE(default, never):Float = 2;
 
-	override public function create()
-	{
+	public static var heroPowerSelection:String = "Aegis";
+
+	override public function create() {
 		super.create();
 
 		player = new Player(FlxG.width / 2, FlxG.height / 2);
+		setUpHeroPower(heroPowerSelection, player);
+		add(heroPower);
 		add(player);
 
 		setUpObstacles();
 		setUpDeadlyObstacles();
 	}
 
-	private function setUpObstacles()
-	{
+	private function setUpHeroPower(heroPowerSelect:String, player:Player) {
+		if (heroPowerSelection == "Invincible") {
+			heroPower = new Invincible(player.x, player.y, player);
+		} else if (heroPowerSelection == "Aegis") {
+			heroPower = new Aegis(player.x, player.y, player);
+		}
```

Comment:
this kind of string comparison is functional, but a bit slow and introduces some risk that a programmer will make some typo somewhere and end up with some broken code.

A situation like this where you have a finite set of valid items / types of a thing is a great spot to use an enum: https://haxe.org/manual/types-enum-using.html 

For us, I think we could have a Power enum (or something like that) defined in its own file, then classes like this could use it for comparisons.

case statements also end up letting you cut out some of the syntax overhead going on, which might also be pleasant.